### PR TITLE
Fix link back to editor from revisions screen

### PR DESCRIPTION
**Pull request by adamsilverstein**
## Description
Fix link back to editor from revisions screen.

* Only show the revisions panel if at least 2 revisions exist. The revisions screen is not designed to display a single revision and is broken in this state. This behavior matches the current core behavior.
* Add a gutenberg query var to the revisions link. This helps identify the correct link to use for returning to the editor from the revisions screen.
* Only reset the classic editor link - if the gutenberg query var is present, use the gurenberg link.

Fixes https://github.com/WordPress/gutenberg/issues/2808

## How Has This Been Tested?
In gutenberg:

* Create post with only title. 
* Change title.
* Go to revisions screen.
* Click return to editor link
Confirm: back in gutenberg

* Create a post with classic editor
* repeat previous steps
Confirm: back to editor link goes back to classic editor.

## Checklist:
- [ ] My code is tested. 
Actually, we could probably add a test to confirm the revisions link only appears when there at least two revisions. I feel like we had this test before.
- [X] My code follows the WordPress code style.
- [X] My code follows has proper inline documentation.

### CODE REVIEW

File: lib/register.php
Comment (aduth) [inline lib/register.php]: Minor: This could be merged into the above condition:

```php
if ( 'revision.php' === $pagenow && ! isset( $_REQUEST['gutenberg'] ) ) {
```

File: lib/register.php
Comment (aduth) [inline lib/register.php]: Should we update the function documentation to reflect that it only redirects to classic editor when it's intended to?

**Review by aduth — APPROVED** [review]

File: lib/register.php
Comment (adamsilverstein) [reply lib/register.php]: yea, good suggestion. i had it that way originally and then thought it might be clearer to break out.

File: lib/register.php
Comment (adamsilverstein) [reply lib/register.php]: yes, i'll add that.

**Comment by jacboy** [comment]
Would this solve [my problem](https://github.com/WordPress/gutenberg/issues/2808#issuecomment-349328611)? How can I apply this to Gutenberg v 1.8.1?

**Comment by aduth** [comment]
Another issue here is that restoring the revision still returns to the classic editor, even when navigating revisions from Gutenberg. I'm going to take a look.

@jacboy I expect it would resolve once merged to master, at which point you'd either need to wait for the next release or run from the repository [using these instructions](https://github.com/WordPress/gutenberg/blob/master/CONTRIBUTING.md).

**Comment by aduth** [comment]
Rebased and pushed with suggested revisions, including a fix for the issue described in https://github.com/WordPress/gutenberg/pull/3511#issuecomment-349365756.

@adamsilverstein Would you have availability to review these latest changes?

**Comment by jacboy** [comment]
@aduth Thanks for your answer. As I can't handle these steps to build the plugin, I kindly want to ask if there is a roadmap for the next release fixing this issue or a workaround on how to get the content back?

**Comment by aduth** [comment]
@jacboy Releases are intended to occur approximately every week, though with WordCamp US this past weekend, it may be until next week before the next release.

**Comment by adamsilverstein** [comment]
@aduth I'll review this today :)

**Comment by adamsilverstein** [comment]
@aduth this tests well locally, going to think about how we can add some test coverage

**Comment by adamsilverstein** [comment]
@aduth I added some phpunit tests to validate the filters we have applied. 

I tried working on jest to test `PostLastRevisionCheck` which should only render if revisionsCount > 1 in this commit - https://github.com/adamsilverstein/gutenberg/commit/a32807c21e5c6120972c4334f712e40c1cc8a260. I got stuck trying to set up the test (I get `Invariant Violation: Could not find "store" in either the context or props of "Connect(PostLastRevisionCheck)". Either wrap the root component in a <Provider>, or explicitly pass "store" as a prop to "Connect(PostLastRevisionCheck)".` because I'm doing it wrong. Any help there appreciated or we can leave that test off.

**Comment by aduth** [comment]
@adamsilverstein It's because the default export of `post-last-revision/index.js` is the [React-Redux connected component](https://github.com/reactjs/react-redux) which is assuming to find a `store` object in context. So you'd either need to provide this context or, more preferably, test the unconnected component and pass in props you might expect to be populated by the `connect`.

You can see an example of this here:

https://github.com/WordPress/gutenberg/blob/ad5db70361b86d5ca9aafcd81fd6d8f040bea177/editor/components/autosave-monitor/index.js#L20

Where in addition to the default export, we also create a separate named export for the class itself, which is used in the tests:

https://github.com/WordPress/gutenberg/blob/ad5db70361b86d5ca9aafcd81fd6d8f040bea177/editor/components/autosave-monitor/test/index.js#L9

**Comment by jacboy** [comment]
1.9.0 has been released, this is still not working. Issue is open so I guess this hasn't been implemented?

**Comment by adamsilverstein** [comment]
@aduth I added some tests by exporting `LastRevision` explicitly, not sure about the exact approach but it does validate the functionality.

**Review by aduth — APPROVED** [review]
Ok, I think I misunderstood how these components were set up in my previous recommendations. For shallow testing, we'd want to ensure we're targeting the specific component we're testing, in this case the `PostLastRevisionCheck` component. So this is the one which should have a named export (from `check.js`). I'm honestly not sure how the previous tests were working, but we shouldn't be reaching into inner component behaviors for shallow testing. 

I pushed updated tests with resolved lint issues in 4c13d2c

**Comment by mtias** [comment]
Thanks for working on this.

**Pull request merged**
